### DISCUSSION TITLE
How to schedule upgrades on default group presets?

### DISCUSSION THREAD

**Discussion by astorije** [renovatebot/renovate#22834] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We try to rely on the default configuration as much as we can. When we cannot, we try to rely on presets as much as we can. We try to configure Renovate to our needs, but while staying as "close to metal" as possible to make future upgrades easier and because, let's be honest, Renovate is amazingly configured by default!

However, some packages are really noisy and create many PRs a week on each of our 50+ Renovate-maintained at the moment. We'd like to reduce the cadence a little.

For example, we want to schedule the [React Router monorepo](https://docs.renovatebot.com/presets-group/#groupreact-routermonorepo) every other week. I don't want to reconfigure the group from scratch, I'd rather extend the existing one.

Is it possible to configure a package rule so to say "use the following schedule string on the `group:react-routerMonorepo`" (which itself is part of the `config:base` preset)?

### What you tried so far.

I think the result would look like something like this, but I'm not sure if it's right, i.e. if it will stop upgrading at-will and use a schedule instead, so I'd love some community help :)

```json
{
  "extends": ["config:base"],
  "packageRules": [
    {
      "extends": ["group:react-routerMonorepo"],
      "schedule": ["every first and third Monday of the month"]
    }
  ]
}
```

(I know that's not exactly bi-weekly, but I'm not sure how to tell Renovate otherwise, because we want to alternate dependencies to reduce lockfile conflicts, i.e. A on first and third Monday, B on second and fourth Monday, etc.)

**Comment by viceice** [comment]
extend the `monorepo:xx` preset instead of the group. so you don't change the group, but match the same packages as the group.

**Reply by RahulGautamSingh** [reply]
For the scheduling you can use this codepen to verify : https://codepen.io/rationaltiger24/full/ZExQEgK

For you use-case you can try something like this: `on the 1st, 3rd week every month`

**Reply by RahulGautamSingh** [reply]
@viceice Should we add `bi-weekly` schedule constant to our docs? Seems like a helpful one to have.

**Reply by astorije-vmware** [reply]
@viceice wouldn't the group preset in `config:base` still apply and keep upgrades at every release, on top of scheduled in our config?

**Reply by viceice** [reply]
no, last package rule wins, so your local rules override any preset rule (only overridden properties of cause)

**Reply by astorije** [reply]
> For the scheduling you can use this codepen to verify : https://codepen.io/rationaltiger24/full/ZExQEgK

Mmh, I was looking for something like that, unfortunately that codepen doesn't seem to work for me, nothing happens when I click validate, whether the string is valid or not. No error in dev console either, just... nothing happens 😅

**Reply by astorije** [reply]
Good to know, I'll try that @viceice, thanks!
I'm a little confused though, why should I extend the `monorepo:xxx` preset instead of the `group:xxx` preset? 🤔

**Reply by viceice** [reply]
because the group extends the monorepo, and you don't need to override the grouping again

**Reply by astorije** [reply]
For the [`react` group](https://docs.renovatebot.com/presets-group/#groupreact), which is not directly mapped to a monorepo, is my best bet to extend using the group name (which is the same strategy used for [the group itself](https://github.com/renovatebot/renovate/pull/20462)):

```json
{
  "groupName": "react monorepo",
  "schedule": ["..."]
}
```

?

EDIT: Welp, that doesn't work...
```
packageRules[3]: Each packageRule must contain at least one match* or exclude* selector. Rule: {\"groupName\":\"react monorepo\",\"schedule\":[\"on the 1st day of the month\"]}
```

**Comment by RahulGautamSingh** [comment]
> > For the scheduling you can use this codepen to verify : https://codepen.io/rationaltiger24/full/ZExQEgK
> 
> Mmh, I was looking for something like that, unfortunately that codepen doesn't seem to work for me, nothing happens when I click validate, whether the string is valid or not. No error in dev console either, just... nothing happens 😅

You need to be logged in🙃

**Reply by astorije** [reply]
🤦‍♂️ Thanks!

**Reply by astorije** [reply]
Ok I feel dumb. Do you know how I can configure it as "on the 1st and 3rd Monday of the month"? (so I can have another rule that says "on the 2nd and 4th Monday of the month") I've tried everything I could think of without success

**Reply by RahulGautamSingh** [reply]
The later syntax maybe doesn't directly support bi-weekly. The syntax I suggested above is valid but might not work. 
I checked on the same site and next occurence doesn't match 😓.

**Reply by RahulGautamSingh** [reply]
The best I could get so far with certainity is:
For package A: `on the first day of the month and on the 15th day of the month` and 
For package B: `on the 8th day of the month and on the 23rd day of the month`

**Reply by astorije** [reply]
That's what I ended up doing 😅 Thank you!

**Reply by viceice** [reply]
isn't this easier to solve with cron syntax?

**Reply by astorije** [reply]
How do you mean?

**Reply by viceice** [reply]
schedule config option also supports cron expression

**Reply by HonkingGoose** [reply]
Related issue:

- #21523
